sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/SimplifiedKinesisClient.java

```diff
@@ -61,18 +69,33 @@
   private static final int PERIOD_GRANULARITY_IN_SECONDS = 60;
   private static final String SUM_STATISTIC = "Sum";
   private static final String STREAM_NAME_DIMENSION = "StreamName";
-  private static final int LIST_SHARDS_DESCRIBE_STREAM_MAX_ATTEMPTS = 10;
-  private static final Duration LIST_SHARDS_DESCRIBE_STREAM_INITIAL_BACKOFF =
+  private static final int LIST_SHARDS_MAX_RESULTS = 1_000;
+  private static final Duration
+      SPACING_FOR_TIMESTAMP_LIST_SHARDS_REQUEST_TO_NOT_EXCEED_TRIM_HORIZON =
+          Duration.standardMinutes(5);
+  private static final int DESCRIBE_STREAM_SUMMARY_MAX_ATTEMPTS = 10;
+  private static final Duration DESCRIBE_STREAM_SUMMARY_INITIAL_BACKOFF =
       Duration.standardSeconds(1);
+
   private final AmazonKinesis kinesis;
   private final AmazonCloudWatch cloudWatch;
   private final Integer limit;
+  private final Supplier<Instant> currentInstantSupplier;
 
   public SimplifiedKinesisClient(
       AmazonKinesis kinesis, AmazonCloudWatch cloudWatch, Integer limit) {
+    this(kinesis, cloudWatch, limit, Instant::now);
```

Comment:
Isn't this a part of public user API that is being changed?

Reply:
This class is package-private, so even if constructor is public it should not be a problem. Constructor doesn't have new parameter, it provides default value. `checkNotNull` calls verifying arguments that were here before are in package-private constructor used for tests.
Could you please pinpoint where API has changed, @aromanenko-dev ?

Reply:
You right, I was mistaken.